src/org/zaproxy/zap/extension/jxbrowserlinux32/ExtensionJxBrowserLinux32.java

```diff
@@ -0,0 +1,141 @@
+
+    private static final ImageIcon CHROMIUM_ICON = DisplayUtils.getScaledIcon(new ImageIcon(
+            ExtensionJxBrowserLinux32.class.getResource(ExtensionJxBrowserLinux32.RESOURCE + "/chromium.png")));
+
+    protected static final String RESOURCE = "/org/zaproxy/zap/extension/jxbrowserlinux32/resources";
+
+    private JButton launchBrowserButton;
+
+    public ExtensionJxBrowserLinux32() {
+        super(NAME);
+    }
+
+    @Override
+    public void init() {
+        super.init();
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        if (getView() != null) {
+
+            if (Constant.isLinux() && ! isOs64Bits()) {
+                // Only show if we're running on the right platform
+                View.getSingleton().addMainToolbarButton(this.getLaunchBrowserButton());
+    
+                ZapMenuItem menulaunch = new ZapMenuItem("jxbrowserlinux32.menu.launch");
+                menulaunch.addActionListener(new java.awt.event.ActionListener() {
+    
+                    @Override
+                    public void actionPerformed(java.awt.event.ActionEvent ae) {
+                        launchBrowser(null);
+                    }
+                });
+    
+                extensionHook.getHookMenu().addToolsMenuItem(menulaunch);
+            }
+        }
+    }
+    
+    private static boolean isOs64Bits() {
+        String arch = System.getProperty("os.arch");
+        return arch.contains("amd64") || arch.contains("x86_64");
+    }
+    
+    private JButton getLaunchBrowserButton() {
+        if (launchBrowserButton == null) {
+            launchBrowserButton = new JButton();
+            launchBrowserButton.setIcon(CHROMIUM_ICON);
+            launchBrowserButton.setToolTipText(Constant.messages.getString("jxbrowserlinux32.button.launch"));
+            launchBrowserButton.addActionListener(new ActionListener() {
+
+                @Override
+                public void actionPerformed(ActionEvent e) {
+                    launchBrowser(null);
+                }
+            });
+        }
+        return launchBrowserButton;
+    }
+
+    private void launchBrowser(final String url) {
+        SwingUtilities.invokeLater(new Runnable() {
+
+            @Override
+            public void run() {
+                try {
+                    ZapBrowserFrame zbf = new ZapBrowserFrame(true, true);
+                    if (url != null) {
+                        zbf.getBrowser().loadURL(url);
+                    }
+                } catch (Exception e) {
+                    LOGGER.error(e.getMessage(), e);
+                }
+            }
+        });
+
+    }
+
+    @Override
+    public boolean canUnload() {
+        return true;
+    }
+
+    @Override
+    public void unload() {
+        if (getView() != null) {
+            if (Constant.isLinux() && ! isOs64Bits()) {

```

Comment:
No, but if it was not added it does not need to be removed, we could replace those checks with just a `launchBrowserButton` null check though (since that seems to be initialised only when added).